src/Queue.js

```diff
@@ -60,21 +62,37 @@ class Queue {
     // If it isn't finished, emit a scheduler interrupt notifying the scheduler that this process
     // needs to be moved to a lower priority queue
     manageTimeSlice(currentProcess, time) {
-
+        if (currentProcess.stateChanged) {
+            this.quantumClock = 0;
+            return;
+        } else {
+            this.quantumClock += time;
+            if (this.quantumClock > this.quantum) {
+                this.quantumClock = 0;
+                let process = this.dequeue();
+                if (process.cpuTimeNeeded) {
+                    SchedulerInterrupt.LOWER_PRIORITY;
+                }
+            }
+        }
```

Comment:
For this block of code there are a couple of things missing/incorrect.
In your first if statement on line 65 you want it be like this because you are calling the function/method stateChanged
`if (currentProcess.isStateChanged())`
the `()` at the end make sure it actually calls it as it's a method from the Process file.
Then on line 14 you need to call the `handleInterrupt` so it would be like this
`this.scheduler.handleInterrupt(this, process, SchedulerInterrupt.LOWER_PRIORITY);`
The rest is great